src/klein/test/typing_app.py

```diff
@@ -0,0 +1,71 @@
+"""
+This file contains code that should validate with type checking if type hints
+are correct.
```

Comment:
Maybe rename this `mypy_checks.py` or something like that and add a note that this code is not expected  to be executed.

Or is this file used outside of mypy checks?

```suggestion
are correct.

The code is not executed, it is here just to be checked by mypy.
```

Reply:
I considered a name like that but:
 * It's type checking, not necessarily specific to mypy as the checker, hence `typing_`
 * These checks are specific to `_app.py` and I thought we might need checks for other files and it might be good to group them in corresponding files. Possibly overkill, but we can change this later.

This is a bit of an odd duck, since I'm looking to verify that the type hints do things statically and not at run time. I'm not sure it's the best solution, but it's working for now.